## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#1a752ddb-ac93-4aa9-b136-5edc964d730d) |
| mint-xl.sh | :x: [more...](#711b7ecf-dbf5-479a-a29c-c14cce9eb3d7) |
| mint-compression-dist-xl.sh | :x: [more...](#0ac581b8-4d04-4bf6-b4e5-3c5e9400895f) |
| mint-gateway-nas.sh | :x: [more...](#2ef9d64c-79a9-4d7b-9520-97a5ba4327c4) |
<a name="711b7ecf-dbf5-479a-a29c-c14cce9eb3d7"></a>
### 8337-c6c3f07/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30214
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3240bac84614:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 54 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 6 seconds
(8/14) Running minio-dotnet tests ... done in 32 seconds
(9/14) Running minio-go tests ... done in 58 seconds
(10/14) Running minio-java tests ... done in 21 seconds
(11/14) Running minio-js tests ... done in 45 seconds
(12/14) Running minio-py tests ... done in 1 minutes and 15 seconds
(13/14) Running s3cmd tests ... FAILED in 9 seconds
{
  "name": "s3cmd",
  "duration": "3964",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-28831/s3cfg get s3://s3cmd-test-bucket-28587/s3cmd-test-object-27934 s3cmd-test-object-27934.downloaded\nProblem: <class 'ConnectionResetError: [Errno 104] Connection reset by peer\nS3cmd:   2.0.2\npython:   3.5.2 (default, Jul 10 2019, 11:58:48) \n[GCC 5.4.0 20160609]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3092, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3001, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 590, in cmd_object_get\n    response = s3.object_get(uri, dst_stream, destination, start_position = start_position, extra_label = seq_label)\n  File \"/usr/local/lib/python3.5/dist-packages/S3/S3.py\", line 716, in object_get\n    response = self.recv_file(request, stream, labels, start_position)\n  File \"/usr/local/lib/python3.5/dist-packages/S3/S3.py\", line 1742, in recv_file\n    data = http_response.read(this_chunk)\n  File \"/usr/lib/python3.5/http/client.py\", line 458, in read\n    n = self.readinto(b)\n  File \"/usr/lib/python3.5/http/client.py\", line 498, in readinto\n    n = self.fp.readinto(b)\n  File \"/usr/lib/python3.5/socket.py\", line 575, in readinto\n    return self._sock.recv_into(b)\nConnectionResetError: [Errno 104] Connection reset by peer\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}

Executed 12 out of 14 tests successfully.
```
<a name="1a752ddb-ac93-4aa9-b136-5edc964d730d"></a>
### 8337-c6c3f07/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:31275
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c364820cbb88:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 58 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 48 seconds
(8/14) Running minio-dotnet tests ... done in 25 seconds
(9/14) Running minio-go tests ... done in 47 seconds
(10/14) Running minio-java tests ... done in 22 seconds
(11/14) Running minio-js tests ... FAILED in 9 seconds
{
  "name": "minio-js",
  "function": "fPutObject(bucketName, objectName, filePath, callback)",
  "args": "bucketName:minio-js-test-cd43b82c-cbfb-4233-93c9-d46df5da7a6b, objectName:datafile-65-MB, filePath:/tmp/datafile-65-MB",
  "duration": 1330,
  "status": "FAIL",
  "error": "Error: read ECONNRESET at exports._errnoException (util.js:1020:11) at TCP.onread (net.js:580:26)"
}

Executed 10 out of 14 tests successfully.
```
<a name="0ac581b8-4d04-4bf6-b4e5-3c5e9400895f"></a>
### 8337-c6c3f07/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:31831
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7ffb61bde140:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 45 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 1 minutes and 6 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 26 seconds
(8/14) Running minio-dotnet tests ... done in 35 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 24 seconds
(10/14) Running minio-java tests ... done in 2 minutes and 5 seconds
(11/14) Running minio-js tests ... FAILED in 18 seconds
{
  "name": "minio-js",
  "function": "fPutObject(bucketName, objectName, filePath, metaData)",
  "args": "bucketName:minio-js-test-f9f3badf-5164-4bfc-a9b5-61a40080cd7e, objectName:datafile-65-MB, filePath:/tmp/datafile-65-MB",
  "duration": 1033,
  "status": "FAIL",
  "error": "Error: read ECONNRESET at exports._errnoException (util.js:1020:11) at TCP.onread (net.js:580:26)"
}

Executed 10 out of 14 tests successfully.
```
<a name="2ef9d64c-79a9-4d7b-9520-97a5ba4327c4"></a>
### 8337-c6c3f07/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:30879
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 48d4bd416f23:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 54 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 48 seconds
(8/14) Running minio-dotnet tests ... done in 27 seconds
(9/14) Running minio-go tests ... done in 57 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 18 seconds
(11/14) Running minio-js tests ... done in 33 seconds
(12/14) Running minio-py tests ... FAILED in 2 minutes and 8 seconds
{
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "args": {
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>d765e83f-2964-4baa-9a15-97105c469e1f</Key><BucketName>minio-py-test-4d278a03-3cc6-474b-8f2e-6313ce21b778</BucketName><Resource>/minio-py-test-4d278a03-3cc6-474b-8f2e-6313ce21b778/d765e83f-2964-4baa-9a15-97105c469e1f</Resource><RequestId>15C94313D85DF1B0</RequestId><HostId></HostId></Error>",
    "object_name": "d765e83f-2964-4baa-9a15-97105c469e1f",
    "bucket_name": "minio-py-test-4d278a03-3cc6-474b-8f2e-6313ce21b778",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"404\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-09-30T15-46-30Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15C94313D85DF1B0\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Mon, 30 Sep 2019 16:08:17 GMT\"]}}",
    "response.status": 403,
    "response.reason": "Forbidden",
    "presigned_get_object_url": "http://72.28.97.51:30879/minio-py-test-4d278a03-3cc6-474b-8f2e-6313ce21b778/d765e83f-2964-4baa-9a15-97105c469e1f?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20190930%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20190930T160810Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=80a1fbfa8a5688ce93eb6023edb2fba4c878732483603496e9bb62a2a3cfba63"
  },
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "duration": 48,
  "message": "AccessDenied: message: Access Denied",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1341, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2036, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1355, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "status": "FAIL"
}

Executed 11 out of 14 tests successfully.
```

--- a/cmd/api-resources.go
+++ b/cmd/api-resources.go
@@ -96,6 +96,7 @@ func getListObjectsV2Args(values url.Values) (prefix, token, startAfter, delimit
 		decodedToken, err := base64.StdEncoding.DecodeString(token)
 		if err != nil {
 			errCode = ErrIncorrectContinuationToken
+			return
 		}
 		token = string(decodedToken)
 	}

